test/SpagoSpec.hs

```diff
@@ -0,0 +1,158 @@
+module SpagoSpec (spec) where
+
+import           Control.Concurrent  (threadDelay)
+import           Control.Monad.Extra (whenM)
+import           Prelude             hiding (FilePath)
+import           Test.Hspec          (Spec, around_, beforeAll, describe, it,
+                                      shouldBe)
+import           Turtle              (FilePath, cp, mkdir, mv, readTextFile, rm,
+                                      rmtree, testdir, writeTextFile)
+import           Utils               (checkFixture, runFor, shouldBeFailure,
+                                      shouldBeFailureOutput, shouldBeSuccess,
+                                      shouldBeSuccessOutput, spago, withCwd)
+
+testDir :: FilePath
+testDir = "test/spago-test"
+
+clean :: IO ()
+clean = do
+  whenM (testdir testDir) (rmtree testDir)
+  mkdir testDir
+
+spec :: Spec
+spec = beforeAll clean $ around_ (withCwd testDir) $ do
+
+  describe "spago init" $ do
+
+    it "Spago should have set up a project" $ do
+
+      spago ["init"] >>= shouldBeSuccess
+
+    it "Spago should refuse to overwrite an existing project without -f" $ do
+
+      spago ["init"] >>= shouldBeFailure
+
+    it "Cleaning of config files should succeed" $ do
```

Comment:
```suggestion
```

If we remove this line the cleanup will be done as part of the previous step, as it happens in other cases too